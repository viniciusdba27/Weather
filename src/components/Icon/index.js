import React from "react";
import { Helpers, Metrics } from "../../theme"; // TODO: add babel
import Config from "../../config";
import Styles from "./styles";

const Icon = ({ item }) => {
  const { applicable_date, weather_state_abbr } = item;
  const url = `${Config.icon}${weather_state_abbr}.png`;
  console.log("ITEM", item);
  return (
    <div style={{ ...Helpers.fillRow, ...Metrics.verticalMargin }}>
      <p>{new Date(applicable_date).toDateString()}</p>
      <div style={Styles.icon(url)} />
    </div>
  );
};

export default Icon;
