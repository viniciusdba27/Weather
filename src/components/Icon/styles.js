const Styles = {
  icon: (img) => ({
    backgroundImage: `url(${img})`,
    width: 64,
    height: 64,
  }),
};
export default Styles;
