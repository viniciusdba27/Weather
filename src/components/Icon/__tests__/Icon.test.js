import React from "react";
import Icon from "../index";
import renderer from "react-test-renderer";

const mockedData = {
  id: 5136007935557632,
  weather_state_name: "Heavy Cloud",
  weather_state_abbr: "hc",
  wind_direction_compass: "WNW",
  created: "2021-11-20T03:59:17.329185Z",
  applicable_date: "2021-11-19",
  min_temp: 12.56,
  max_temp: 15.44,
  the_temp: 16.195,
  wind_speed: 3.6720387739070492,
  wind_direction: 287.57292905230133,
  air_pressure: 1021.5,
  humidity: 77,
  visibility: 6.242294997216257,
  predictability: 71,
};

it("Icon renders correctly", async () => {
  const component = renderer.create(<Icon item={mockedData} />).toJSON();
  expect(component).toMatchSnapshot();
});
