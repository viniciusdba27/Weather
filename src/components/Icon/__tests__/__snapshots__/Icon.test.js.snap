// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<div
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "marginBottom": 15,
      "marginTop": 15,
    }
  }
>
  <p>
    Thu Nov 18 2021
  </p>
  <div
    style={
      Object {
        "backgroundImage": "url(https://www.metaweather.com/static/img/weather/png/64/hc.png)",
        "height": 64,
        "width": 64,
      }
    }
  />
</div>
`;
